### PR TITLE
Fix Component governance bugs

### PR DESCRIPTION
**Pull request by wtgodbe**

### CODE REVIEW

**Comment by wtgodbe** [comment]
> Could not find AzureStorageEmulator.exe. Please install Azure SDK from http://azure.microsoft.com/en-us/downloads/

@MattGal @dotnet/dnceng the build used to work on these machines, have they changed? https://github.com/aspnet/AspNetWebHooks/blob/5f384bda295a112b0788dad5992376d1781857ee/.azure-ci.yml#L33-L34

**Comment by MattGal** [comment]
> > Could not find AzureStorageEmulator.exe. Please install Azure SDK from http://azure.microsoft.com/en-us/downloads/
> 
> @MattGal @dotnet/dnceng the build used to work on these machines, have they changed?
> 
> https://github.com/aspnet/AspNetWebHooks/blob/5f384bda295a112b0788dad5992376d1781857ee/.azure-ci.yml#L33-L34


Taking a look, it looks like @dnceng tagging isn't working in this repo for some reason.

**Comment by MattGal** [comment]
@wtgodbe:

You can get info about what changes in  'windows-latest' from commit history [here](https://github.com/actions/virtual-environments).  Most recent thing that I know of is moving from 2019 being default to 2022. 

To the un-asked questions, there's two things going on here.  
- In general, you need to stop using AzDO Hosted builds
- The pool provider mentioned in your yaml is long dead (I'm thinking this build hasn't run in some time?)

So assuming you wanted to address both (feel free to do either/or; some folks may talk to you about continued hosted usage though), it'd be something like going from:
  
```
    pool:
    vmImage: windows-latest
    ${{ if ne(variables['System.TeamProject'], 'public') }}:
      # This override makes the specified vmImage irrelevant.
      name: NetCoreInternal-Pool
      queue: BuildPool.Server.Amd64.VS2017
```	  
... to: 
```
    pool:
    ${{ if eq(variables['System.TeamProject'], 'public') }}:
      name: NetCore1ESPool-Public
      demands: ImageOverride -equals build.windows.10.amd64.vs2017.open
    ${{ if ne(variables['System.TeamProject'], 'public') }}:
      name: NetCore1ESPool-Internal
      demands: ImageOverride -equals build.windows.10.amd64.vs2017
```	
 
All available 1ES Hosted build images are listed in https://helix.dot.net/, I've picked VS 2017 ones since that seems to be what you had before.  We also mirror the "Azure hosted" VS 2019 and VS 2022 images (via `1es-windows-2019`/`1es-windows-2022`) if VS 2019 is acceptable ; latest is VS 2022 for sure.

**Comment by wtgodbe** [comment]
Thanks @MattGal - trying that now

File: .azure-ci.yml [hunk 1]
```diff
@@ -31,11 +31,12 @@ jobs:
   workspace:
     clean: all
   pool:
-    vmImage: windows-latest
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      name: NetCore1ESPool-Public
+      demands: ImageOverride -equals build.windows.10.amd64.vs2017.open
```
Comment (dougbu) [inline .azure-ci.yml:36]: @MattGal I thought we moved off anything using VS 2017 a while back. Was that move not required❔

**Review by dougbu — APPROVED** [review]

File: .azure-ci.yml (hunk 1 above)
Comment (MattGal) [reply .azure-ci.yml:36]: > @MattGal I thought we moved off anything using VS 2017 a while back. Was that move not required❔

If I said that, I was mistaken (feel free to point me at the place causing this confusion and I'll try to clarify).  We've been pushing you to get off of hosted (1ES requirement), and we deleted the Helix "arcade-pool-provider" (1ES requirement also), but in general if we're still putting images up on helix.dot.net those images are good to use until they're no longer there.

**Pull request merged**